challenge_receptionist/src/challenge_receptionist/learn_guest.py

```diff
@@ -101,14 +102,25 @@ def __init__(self, robot,
 
             smach.StateMachine.add('HEAR_DRINK_ANSWER',
                                    HearOptionsExtra(robot,
-                                                           self.drink_spec_des,
-                                                           guest_drink_des.writeable),
-                                   transitions={'heard': 'SAY_DRINK_CORRECT_NAME',
+                                                    self.drink_spec_des,
+                                                    guest_drink_hmi_des.writeable),
+                                   transitions={'heard': 'WRITE_DRINK_NAME',
                                                 'no_result': 'SAY_DRINK_QUESTION'})
 
+            @cb_interface(outcomes=['done'])
+            def _write_drink_name(ud, answer, drink_name_designator):
+                name = answer.resolve()
+                drink_name_designator.write(name)
+                return 'done'
+
+            smach.StateMachine.add("WRITE_DRINK_NAME",
+                                   CBState(_write_drink_name,
+                                           cb_args=[guest_drink_name_des, guest_drink_des.writeable]),
+                                   transitions={'done': 'SAY_DRINK_CORRECT_NAME'})
```

Comment:
What is happening here? We get the value of one designator and write it to another one?